File: collect_app/src/main/java/org/odk/collect/android/instancemanagement/SubmitException.java
package org.odk.collect.android.instancemanagement;

public class SubmitException extends Exception {

    public enum Type {
        GOOGLE_ACCOUNT_NOT_SET,
        GOOGLE_ACCOUNT_NOT_PERMITTED,
        NOTHING_TO_SUBMIT;
    }

    private final Type type;

    public SubmitException(Type type) {
        this.type = type;
    }

    public Type getType() {
        return type;
    }
}

File: app/src/main/java/org/odk/collect/android/instancemanagement/SubmitException.kt
package org.odk.collect.android.instancemanagement

import java.lang.Exception

class SubmitException(val type: Type) : Exception() {
    enum class Type {
        GOOGLE_ACCOUNT_NOT_SET,
        GOOGLE_ACCOUNT_NOT_PERMITTED,
        NOTHING_TO_SUBMIT
    }
}
